src/MiniProfiler.Shared/SqlFormatters/InlineFormatter.cs

```diff
@@ -35,22 +34,21 @@ public string FormatSql(string commandText, List<SqlTimingParameter> parameters)
                 return commandText;
             }
 
-            var originalCommandText = commandText;
-
+            var paramValuesByName = new Dictionary<string, string>(parameters.Count);
             foreach (var p in parameters)
             {
-                // If the parameter doesn't have a prefix (@,:,etc), append one
-                var name = ParamPrefixes.IsMatch(p.Name)
-                    ? p.Name
-                    : Regex.Match(originalCommandText, "([@:?])" + Regex.Escape(p.Name), RegexOptions.IgnoreCase).Value;
-                if (name.HasValue())
-                {
-                    var value = GetParameterValue(p);
-                    commandText = Regex.Replace(commandText, "(" + Regex.Escape(name) + ")([^0-9A-z]|$)", m => value + m.Groups[2], RegexOptions.IgnoreCase);
-                }
+                var trimmedName = p.Name.TrimStart('@', ':', '?').ToLower();
+                paramValuesByName[trimmedName] = GetParameterValue(p);
             }
 
-            return commandText;
+            var regexPattern = "[@:?](?:" + string.Join("|", paramValuesByName.Keys.Select(Regex.Escape)) + ")(?![0-9a-z])";
+
+            return Regex.Replace(
+                commandText,
+                regexPattern,
+                m => paramValuesByName[m.Value.Substring(1).ToLower()],
```

Comment:
Capturing groups are relatively expensive for .NET regex, and benchmarking showed that doing it this way would drop allocations by ~20%.